rpc/types_block.go

```diff
@@ -27,14 +27,13 @@ type BlockID struct {
 	Tag    string     `json:"block_tag,omitempty"`
 }
 
-// Used in the 'starknet_getStorageProof' RPC method, as it doesn't support pending blocks
-type BlockIDWithoutPending BlockID
-
-// Block hash, number or tag, same as BLOCK_ID, but without 'pending'
-type SubscriptionBlockID struct {
-	Number uint64     `json:"block_number,omitempty"`
-	Hash   *felt.Felt `json:"block_hash,omitempty"`
-	Tag    string     `json:"block_tag,omitempty"`
+// checkForPending checks if the block ID has the 'pending' tag. If it does, it returns an error.
+// This is used to prevent the user from using the 'pending' tag on methods that do not support it.
```

Comment:
*nit: does not support it.
Why are all those types removed? Again, Im lacking context a bit